### PR TITLE
bullet: Sync with upstream 3.17

### LINKED ISSUE #43868: [Bullet] [Needs Bullet Update] Crashing asserts in debug when using Bullet Physics with CCD enabled
**Godot version:**
Master 50db0e66ac8c00f176c706cee0c3a5469453acce

**OS/device including version:**
Windows 10

**Issue description:**
Bullet raises some asserts that cause crashes in some scenarios when using continuous collision with 3D rigid bodies.

This issue only happens on Godot 4.0, in debug or release_debug configuration, because these asserts have been enabled with PR #34236.

It's actually happening even when continuous collision is disabled at the moment, until PR #43801 is merged (it fixes Bullet bodies using continuous collision when they shouldn't).

More details in https://github.com/godotengine/godot/pull/42643#issuecomment-732296444.

**Steps to reproduce:**
- Open MRP
- Start game
- Wait for a moment for rigid bodies to be created and observe assert and crash

**Minimal reproduction project:**
[bullet-ccd-4.0.zip](https://github.com/godotengine/godot/files/5599457/bullet-ccd-4.0.zip)

### PR DESCRIPTION
**Pull request by akien-mga**
Includes https://github.com/bulletphysics/bullet3/pull/3799.

Fixes #43868.

The duplicated comments in bullet code are a partial revert of #48187 since I synced with the unchanged `3.17` upstream code. They should be removed upstream if they matter.

---

Note: The bullet module in the `master` branch doesn't actually build currently. This PR just serves to keep the code in sync with the `3.x` branch for future use.

### CODE REVIEW

File: modules/bullet/SCsub [hunk 1]
```diff
@@ -187,6 +187,7 @@ if env["builtin_bullet"]:
         "LinearMath/btGeometryUtil.cpp",
         "LinearMath/btPolarDecomposition.cpp",
         "LinearMath/btQuickprof.cpp",
+        "LinearMath/btReducedVector.cpp",
```
Comment (akien-mga) [inline modules/bullet/SCsub:190]: It seems like you had omitted this `.cpp` file when updating to 2.90/master a while ago: 3e7db60d56d5c25d7aa3fded4b90f36ca341159c.

It built and seemed to work fine without it so maybe it's not needed, it seems to be used in:
```
BulletSoftBody/btDeformableContactProjection.cpp
190:    btReducedVector p(x.size());
```
But in the `#else` of an `#ifndef USE_MGS` (i.e. only used if we pass `USE_MGS`).

I'm adding it for consistency's sake, it should be optimized out by the linker if we don't actually need it I suppose. There's probably a lot of Bullet components which we wouldn't need to build if we don't use them directly, but not sure how easy that would be to identify to slim down our `SCsub`.

**Comment by akien-mga** [comment]
Marking as draft for now as the `3.x` backport leads to a crash: https://github.com/godotengine/godot/pull/48300#issuecomment-829344974

And I just remembered that bullet is disabled on `master` so it doesn't make sense to merge this before #47508 is merged (then I'll rebase and we can confirm that the 3.08 -> 3.09 update is fine).

**Comment by akien-mga** [comment]
Synced with 3.17 which was released recently: https://github.com/bulletphysics/bullet3/releases/tag/3.17
(Don't trust the release date on GitHub, the actual tag is from 7 May 2021.)

**Comment by pouleyKetchoupp** [comment]
Now that #48300 is merged, it would be great to update this PR with the same changes in order to keep master in sync, even if Bullet is disabled for now.

**Comment by akien-mga** [comment]
Synced with #48300, and includes a cherry-pick of #53183.

**Pull request merged**